### PR TITLE
*: decouple FlashGrpcServerHolder from Server.cpp

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #4609 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Server/Server.cpp [hunk 1]
```diff
@@ -297,25 +296,6 @@ pingcap::ClusterConfig getClusterConfig(const TiFlashSecurityConfig & security_c
     return config;
 }
 
-LoggerPtr grpc_log;
```
Comment (ywqzzy) [inline dbms/src/Server/Server.cpp:300]: I removed grpc_log, since it only used once.

File: dbms/src/Server/FlashGrpcServerHolder.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+#include <Server/RaftConfigParser.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+extern const int IP_ADDRESS_NOT_ALLOWED;
+} // namespace ErrorCodes
+
+void handleRpcs(grpc::ServerCompletionQueue * curcq, const LoggerPtr & log);
```
Comment (SeaRise) [inline dbms/src/Server/FlashGrpcServerHolder.h:30]: Is this function referenced by other c++ files?

**Comment by ywqzzy** [comment]
I will move it into anonymous namespace



---Original---
From: ***@***.***&gt;
Date: Mon, Aug 1, 2022 18:22 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [pingcap/tiflash] *: decouple FlashGrpcServerHolder from Server.cpp (PR #5516)




 
@SeaRise commented on this pull request.
 
 
In dbms/src/Server/FlashGrpcServerHolder.h:
 &gt; + +#include <Common/assert_cast.h&gt; +#include <Debug/astToExecutor.h&gt; +#include <Flash/DiagnosticsService.h&gt; +#include <Flash/FlashService.h&gt; +#include <Flash/Mpp/GRPCCompletionQueuePool.h&gt; +#include <Server/RaftConfigParser.h&gt; + +namespace DB +{ +namespace ErrorCodes +{ +extern const int IP_ADDRESS_NOT_ALLOWED; +} // namespace ErrorCodes + +void handleRpcs(grpc::ServerCompletionQueue * curcq, const LoggerPtr &amp; log);  
Is this function referenced by other c++ files?

File: dbms/src/Server/FlashGrpcServerHolder.h (hunk 2 above)
Comment (ywqzzy) [reply dbms/src/Server/FlashGrpcServerHolder.h]: I have moved it.

File: dbms/src/Server/FlashGrpcServerHolder.h [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+#include <Flash/FlashService.h>
+#include <Flash/Mpp/GRPCCompletionQueuePool.h>
+#include <Server/RaftConfigParser.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+extern const int IP_ADDRESS_NOT_ALLOWED;
+} // namespace ErrorCodes
```
Comment (SeaRise) [inline dbms/src/Server/FlashGrpcServerHolder.h:28]: Seems like it should be moved to `FlashGrpcServerHolder.cpp`

**Review by SeaRise — APPROVED** [review]
LGTM

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**